app/controllers/provider_interface/reasons_for_rejection_controller.rb

```diff
@@ -134,5 +135,13 @@ def offer_store
       key = "offer_wizard_store_#{current_provider_user.id}_#{@application_choice.id}"
       WizardStateStores::RedisStore.new(key: key)
     end
+
+    def check_application_is_rejectable
+      return if ApplicationStateChange.new(@application_choice).can_reject?
+
+      flash[:warning] = 'This application cannot be rejected'
```

Comment:
I don't think we want to render a message (based on [Offers](https://github.com/DFE-Digital/apply-for-teacher-training/blob/master/app/controllers/provider_interface/offers_controller.rb#L76) and [Interview](https://github.com/DFE-Digital/apply-for-teacher-training/blob/master/app/controllers/provider_interface/interviews_controller.rb#L160)) but maybe verify with design (cc @adamsilver @johndoates).

Reply:
I would suggest using the standard 404.

Reply:
Thanks for raising this @despo , having spoken with @adamsilver we think a 404 is appropriate. Amended here https://github.com/DFE-Digital/apply-for-teacher-training/pull/4574/commits/f9b5acd1a67923e01c013bfce72d5e6b4c7f925c